test/e2e/deployment_test.go

```diff
@@ -0,0 +1,148 @@
+		Sriov:  &opv1alpha1.Sriov{},
+		Multus: &opv1alpha1.Multus{},
+	}
+	testDeploy(t, configSpec, []Component{SriovComponent, MultusComponent})
+}
+
+func TestDeployNMState(t *testing.T) {
+	configSpec := opv1alpha1.NetworkAddonsConfigSpec{
+		NMState: &opv1alpha1.NMState{},
+	}
+	testDeploy(t, configSpec, []Component{NMStateComponent})
+}
+
+func TestDeployKubeMacPool(t *testing.T) {
+	configSpec := opv1alpha1.NetworkAddonsConfigSpec{
+		KubeMacPool: &opv1alpha1.KubeMacPool{},
+	}
+	testDeploy(t, configSpec, []Component{KubeMacPoolComponent})
+}
+
+func TestDeployAllAtOnce(t *testing.T) {
+	components := []Component{
+		MultusComponent,
+		LinuxBridgeComponent,
+		KubeMacPoolComponent,
+		SriovComponent,
+		NMStateComponent,
+	}
+	configSpec := opv1alpha1.NetworkAddonsConfigSpec{
+		ImagePullPolicy: v1.PullAlways,
+		KubeMacPool:     &opv1alpha1.KubeMacPool{},
+		LinuxBridge:     &opv1alpha1.LinuxBridge{},
+		Multus:          &opv1alpha1.Multus{},
+		Sriov:           &opv1alpha1.Sriov{},
+		NMState:         &opv1alpha1.NMState{},
+	}
+	testDeploy(t, configSpec, components)
+}
+
+func testDeploy(t *testing.T, configSpec opv1alpha1.NetworkAddonsConfigSpec, components []Component) {
+	ctx := prepare(t, components)
+	defer ctx.Cleanup()
+
+	testConfigCreate(t, configSpec, components)
+}
+
+func TestDeployOneByOne(t *testing.T) {
+	ctx := prepare(t, AllComponents)
+	defer ctx.Cleanup()
+
+	configSpec := opv1alpha1.NetworkAddonsConfigSpec{}
+	components := []Component{}
+
+	// Deploy initial empty config
+	testConfigCreate(t, configSpec, components)
+
+	// Deploy Multus component
+	configSpec.Multus = &opv1alpha1.Multus{}
+	components = append(components, MultusComponent)
+	testConfigUpdate(t, configSpec, components)
+
+	// Add Linux bridge component
+	configSpec.LinuxBridge = &opv1alpha1.LinuxBridge{}
+	components = append(components, LinuxBridgeComponent)
+	testConfigUpdate(t, configSpec, components)
+
+	// Add KubeMacPool component
+	configSpec.KubeMacPool = &opv1alpha1.KubeMacPool{}
+	components = append(components, KubeMacPoolComponent)
+	testConfigUpdate(t, configSpec, components)
+
+	// Add SR-IOV component
+	configSpec.Sriov = &opv1alpha1.Sriov{}
+	components = append(components, SriovComponent)
+	testConfigUpdate(t, configSpec, components)
+
+	// Add NMState component
+	configSpec.NMState = &opv1alpha1.NMState{}
+	components = append(components, NMStateComponent)
+	testConfigUpdate(t, configSpec, components)
+}
+
+func testConfigCreate(t *testing.T, configSpec opv1alpha1.NetworkAddonsConfigSpec, components []Component) {
+	// Deploy given Config Spec
+	Assert(t, CreateConfig(configSpec))
+
+	// If there are any components to deploy wait until Progressing condition is reported
+	if len(components) > 0 {
+		Assert(t, CheckConfigCondition(ConditionProgressing, ConditionTrue, time.Minute))
+	}
+
+	// Wait until Available condition is reported
+	Assert(t, CheckConfigCondition(ConditionAvailable, ConditionTrue, 20*time.Minute))
```

Comment:
wow, 20 minutes? can we maybe narrow it down a bit? do we really need all 20 full minutes?

Reply:
It happens so we first run tests pulling individual components, so we load all images on hosts in advance. But in case someone runs only the DeployAllComponents test, it can take a while. However, not 20 minutes, I will set 10, just to be on the safe side while running on my parents' WiFi.

Reply:
maybe worth to mention in a comment here then your thinking

Reply:
Done